src/main/scala/com/rawlabs/das/server/DASSdkManager.scala

```diff
@@ -91,12 +108,16 @@ class DASSdkManager(implicit settings: RawSettings) extends StrictLogging {
    */
   def unregisterDAS(dasId: DASId): Unit = {
     dasSdksInMemoryLock.synchronized {
-      if (dasSdksInMemory.contains(dasId)) {
-        logger.debug(s"Unregistering DAS with ID: $dasId")
-        dasSdksInMemory.remove(dasId)
-        logger.debug(s"DAS unregistered successfully with ID: $dasId")
-      } else {
-        logger.warn(s"Tried to unregister DAS with ID: $dasId, but it was not found.")
+      dasSdksInMemory.get(dasId) match {
+        case Some(DaSDKInMemoryEntry(options, _)) =>
+          logger.debug(s"Unregistering DAS with ID: $dasId")
+          val strippedOptions = options.filterKeys(!_.startsWith("das_"))
+          dasSdksPerConfig.remove(strippedOptions)
```

Comment:
I think you should check if it's the last DAS Id using the SDK, and only remove it if that's the case. Since we're under a lock, I think we're safe (?)

Reply:
BTW, I noticed we hold the lock while initializing the SDK. Please double check if that's necessary (I think so but it blocks most other interactions, as they probably start by obtaining the SDK?

Reply:
I tried a version that uses a Cache/CacheLoader (config => sdk) and still maintains the HashMap per ID to control which config is associated to what. I believe with a version like that, the cache loader only locks the entry of the matching config (it wouldn't block other registrations) and the HashMap isn't locked for long.

Reply:
It isn't called for now. It could be exercised by a test.